src/jsrm/systems/utils.py

```diff
@@ -124,3 +124,86 @@ def compute_planar_stiffness_matrix(l: Array, A: Array, Ib: Array, E: Array, G:
     S = l * jnp.diag(jnp.stack([Ib * E, 4 / 3 * A * G, A * E], axis=0))
 
     return S
+
+def gauss_quadrature(
+    N_GQ: int, 
+    a = 0.0,
+    b = 1.0
+)-> Tuple[Array, Array, int]:
+    """
+    Computes the Legendre-Gauss nodes and weights on the interval [0, 1] 
+    using Legendre-Gauss Quadrature with truncation order N_GQ.
+
+    Args:
+        N_GQ (int): order of the truncature.
+        a (float, optional): The lower bound of the interval. Default is 0.0.
+        b (float, optional): The upper bound of the interval. Default is 1.0.
+
+    Returns:
+        tuple: A tuple containing:
+            - Xs (Array): The Gauss nodes on [a, b].
+            - Ws (Array): The Gauss weights on [a, b].
+            - nGauss (int): The number of Gauss points including boundary points, i.e., N_GQ + 2.
+    """
+    
+    N = N_GQ - 1
+    N1 = N + 1
+    N2 = N + 2
+
+    xu = jnp.linspace(-1, 1, N1)
+
+    # Initial guess
+    y = jnp.cos((2 * jnp.arange(N + 1) + 1) * jnp.pi / (2 * N + 2)) + (0.27 / N1) * jnp.sin(jnp.pi * xu * N / N2)
+
+    def legendre_iteration(y):
+        #Legendre-Gauss Vandermonde matrix
+        L = jnp.zeros((N1, N2))
+        L = L.at[:, 0].set(1)
+        L = L.at[:, 1].set(y)
+
+        for k in range(2, N1 + 1):
+            L = L.at[:, k].set(((2 * k - 1) * y * L[:, k - 1] - (k - 1) * L[:, k - 2]) / k)
```

Comment:
@solangegbv Please see if using concatenation/stacking instead of `.at[].set()` increases performance